homeassistant/components/media_source/local_source.py

```diff
@@ -71,17 +73,42 @@ async def async_browse_media(
     ) -> BrowseMediaSource:
         """Return media."""
         try:
-            source_dir_id, location = async_parse_identifier(item)
+            source_dir_id, location = self.async_parse_identifier(item)
         except Unresolvable as err:
             raise BrowseError(str(err)) from err
 
         return await self.hass.async_add_executor_job(
             self._browse_media, source_dir_id, location
         )
 
-    def _browse_media(self, source_dir_id, location):
+    def _browse_media(self, source_dir_id: str, location: Path):
         """Browse media."""
-        full_path = Path(self.hass.config.path("media", location))
+
+        # If only one media dir is configured, use that as the local media root
+        if source_dir_id == "" and len(self.hass.config.media_dirs) == 1:
+            source_dir_id = list(self.hass.config.media_dirs)[0]
+
+        # Multiple folder, root is requested
+        if source_dir_id == "":
+            base = BrowseMediaSource(
```

Comment:
We should raise if a location is passed in.

```suggestion
            if location:
                raise BrowseError("Folder not found.")

            base = BrowseMediaSource(
```